pkg/manager/member/scaler.go

```diff
@@ -104,3 +113,91 @@ func ordinalPVCName(memberType v1alpha1.MemberType, setName string, ordinal int3
 func ordinalPodName(memberType v1alpha1.MemberType, tcName string, ordinal int32) string {
 	return fmt.Sprintf("%s-%s-%d", tcName, memberType, ordinal)
 }
+
+func validateScaling(actual *apps.StatefulSet, desired *apps.StatefulSet) error {
+	if !features.DefaultFeatureGate.Enabled(features.AdvancedStatefulSet) {
+		return nil
+	}
+	// Following cases are not supported yet.
+	// TODO support it or validate in validation phase
```

Comment:
some scaling operations are not allowed because it has to delete/create pods. I decided to implement in the future. There are two ways to solve this issue:

1) avoid in validation phase
2) support it by scaling out first and then scaling in, e.g.
   -`replicas 5, delete slots [0]` is changed to `replicas 4, delete slots[1]`, we need to delete pod 1 and 5 and create pod 4.